embassy-imxrt/src/gpio.rs

```diff
@@ -0,0 +1,1058 @@
+                    // There is not currently a "safe" method for setting a single-bit.
+                    unsafe { w.dirclrp().bits(1 << self.pin.pin()) });
+    }
+
+    /// Converts pin to special function pin
+    /// # Safety
+    /// Unsafe to require justifying change from default to a special function
+    ///
+    pub unsafe fn set_as_special_function(&mut self, func: Function) {
+        self.pin.set_function(func);
+    }
+
+    /// Is high?
+    #[must_use]
+    pub fn is_high(&self) -> bool {
+        !self.is_low()
+    }
+
+    /// Is low?
+    #[must_use]
+    pub fn is_low(&self) -> bool {
+        self.pin.block().b(self.pin.port()).b_(self.pin.pin()).read() == 0
+    }
+
+    /// Current level
+    #[must_use]
+    pub fn get_level(&self) -> Level {
+        self.is_high().into()
+    }
+
+    /// Wait until the pin is high. If it is already high, return immediately.
+    #[inline]
+    pub async fn wait_for_high(&mut self) {
+        InputFuture::new(self.pin.reborrow(), InterruptType::Level, Level::High).await;
+    }
+
+    /// Wait until the pin is low. If it is already low, return immediately.
+    #[inline]
+    pub async fn wait_for_low(&mut self) {
+        InputFuture::new(self.pin.reborrow(), InterruptType::Level, Level::Low).await;
+    }
+
+    /// Wait for the pin to undergo a transition from low to high.
+    #[inline]
+    pub async fn wait_for_rising_edge(&mut self) {
+        InputFuture::new(self.pin.reborrow(), InterruptType::Edge, Level::High).await;
+    }
+
+    /// Wait for the pin to undergo a transition from high to low.
+    #[inline]
+    pub async fn wait_for_falling_edge(&mut self) {
+        InputFuture::new(self.pin.reborrow(), InterruptType::Edge, Level::Low).await;
+    }
+
+    /// Wait for the pin to undergo any transition, i.e low to high OR high to low.
+    #[inline]
+    pub async fn wait_for_any_edge(&mut self) {
+        if self.is_high() {
+            InputFuture::new(self.pin.reborrow(), InterruptType::Edge, Level::Low).await;
+        } else {
+            InputFuture::new(self.pin.reborrow(), InterruptType::Edge, Level::High).await;
+        }
+    }
+
+    /// Return a new Flex pin instance with level sensing disabled.
+    ///
+    /// Consumes less power than a flex pin with sensing enabled.
+    #[must_use]
+    pub fn disable_sensing(self) -> Flex<'d, SenseDisabled> {
+        // Cloning the pin is ok since we consume self immediately
+        let new_pin = unsafe { self.pin.clone_unchecked() };
+        drop(self);
+        Flex::<SenseDisabled>::new(new_pin)
+    }
+}
+
+impl<'d> Flex<'d, SenseDisabled> {
+    /// New flex pin.
+    pub fn new(pin: Peri<'d, impl GpioPin>) -> Self {
+        pin.set_function(Function::F0)
+            .disable_analog_multiplex()
+            .disable_input_buffer();
+
+        Self {
+            pin: pin.into(),
+            _sense_mode: PhantomData::<SenseDisabled>,
+        }
+    }
```

Comment:
From what I can tell, since both `Flex<SenseDisabled>` and `Flex<SenseEnabled>` both use `new` for the constructor, you will need to specify with a turbofish operator which once you want.

I'd prefer to make it explicit in the function name which mode you have seleted. So a `new_with_sense` for `SenseEnabled` and `new` for `SenseDisabled`.

Reply:
Fixed